numba/np/arraymath.py

```diff
@@ -1474,8 +1474,9 @@ def nan_aware_less_than(a, b):
 
 
 def _partition_factory(pivotimpl):
-    def _partition(A, low, high):
+    def _partition(A, low, high, I=np.arange(0)):
         mid = (low + high) >> 1
+        argpartition = True if len(I) else False
```

Comment:
```suggestion
        argpartition = True if I is None else False
        if argpartition:
            assert len(I) == len(A) # The corresponding index array should be valid for the array provided.
```

Just to make sure every upstream function uses the factory correctly.